### PR TITLE
Pin message video audio attachments

### PR DESCRIPTION
**Pull request by veken1199**
Pin full functionality for normal conversations and group chat

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -108,6 +108,7 @@
   private View                        composeDivider;
   private View                        scrollToBottomButton;
   private TextView                    scrollDateHeader;
+  private int x                       =1;
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Is this being used anywhere? I can't find it.

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 2]
```diff
@@ -84,17 +90,37 @@ public ViewHolder(View v) {
...
+                    = conversationItem.new ThumbnailClickListener(record);
+            ConversationItem.AttachmentDownloadClickListener attachmentDownloadClickListener
+                    = conversationItem.new AttachmentDownloadClickListener();
+            holder.documentViewStub.get().setDocument(((MediaMmsMessageRecord)record).getSlideDeck().getDocumentSlide(), true);
+            holder.documentViewStub.get().setDocumentClickListener(thumbnailClickListener);
+            holder.documentViewStub.get().setDownloadClickListener(attachmentDownloadClickListener);
+            holder.documentViewStub.get().setVisibility(view.VISIBLE);
+        }
+
+        if (isAudio(record)) {
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: Should this one also be `else if`?

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 3]
```diff
@@ -84,17 +90,37 @@ public ViewHolder(View v) {
...
+            ConversationItem.AttachmentDownloadClickListener attachmentDownloadClickListener
+                    = conversationItem.new AttachmentDownloadClickListener();
+            holder.audioViewStub.get().setAudio(masterSecret, ((MediaMmsMessageRecord)record).getSlideDeck().getAudioSlide(), true);
+            holder.audioViewStub.get().setDownloadClickListener(attachmentDownloadClickListener);
+            holder.audioViewStub.get().setVisibility(view.VISIBLE);
+        }
+
+        else if (isVideo(record)) {
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: IMO this should be on the same line as the closing bracket.

https://github.com/signalapp/Signal-Android/search?utf8=%E2%9C%93&q=else+if&type=Code

**Review by g-harel — DISMISSED** [review]
Also should we test this behavior?

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 1 above)
Comment (veken1199) [reply src/org/thoughtcrime/securesms/ConversationFragment.java]: this has slipped

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java (hunk 3 above)
Comment (veken1199) [reply src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: interesting
same goes for try catch

**Review by g-harel — APPROVED** [review]
:+1:

**Pull request merged**